packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
         })
     }
 
-    // TODO: Add functionality for the three checks
-    public async validatePolicy(documentType: string, policyType: string) {
-        this.client /* Do something */
+    /* Uses A2 SDK and Python CLI tools to call ValidatePolicy API
+     * Responses are exposed in the Webview, including errors
+     * Diagnostic objects are created to expose findings, which appear in the problems panel
+     */
+    public async validatePolicy(documentType: string, policyType: string, cfnParameterPath?: string) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        validatePolicyDiagnosticCollection.clear()
+        const diagnostics: vscode.Diagnostic[] = []
+        switch (documentType) {
+            case PolicyChecksDocumentType.JSON_POLICY_LANGUAGE: {
+                this.client.validatePolicy(
+                    {
+                        policyDocument: IamPolicyChecksWebview.editedDocument,
+                        policyType: policyType === 'Identity' ? 'IDENTITY_POLICY' : 'RESOURCE_POLICY',
+                    },
+                    (err, data) => {
+                        if (err) {
+                            this.onValidatePolicyResponse.fire([err.message, 'red'])
+                        } else {
+                            if (data.findings.length > 0) {
+                                data.findings.forEach((finding: AccessAnalyzer.ValidatePolicyFinding) => {
+                                    const message = `${finding.findingType}: ${finding.issueCode} - ${finding.findingDetails} Learn more: ${finding.learnMoreLink}`
+                                    if (finding.findingType === 'ERROR') {
```

Comment:
Is `ERROR` not an enum or something similar?

Reply:
I can create an enum, but since the type is a string from the API response, it will not be strictly respected.

Reply:
yeah, my point is we do not want to copy this string around for comparisons. easy to make typo, and imagine one day that findingType is becoming something else, how many places you'd need to update?